pkg/scheduler/algorithm/priorities/resource_allocation.go

```diff
@@ -91,13 +93,138 @@ func (r *ResourceAllocationPriority) PriorityMap(
 	}, nil
 }
 
-func getNonZeroRequests(pod *v1.Pod) *schedulernodeinfo.Resource {
-	result := &schedulernodeinfo.Resource{}
-	for i := range pod.Spec.Containers {
-		container := &pod.Spec.Containers[i]
-		cpu, memory := priorityutil.GetNonzeroRequests(&container.Resources.Requests)
-		result.MilliCPU += cpu
-		result.Memory += memory
+// calculateResourceAllocableRequest returns resources Allocatable and Requested values
+func calculateResourceAllocableRequest(nodeInfo *schedulernodeinfo.NodeInfo, pod *v1.Pod, resource v1.ResourceName) (int64, int64) {
+	allocatable := nodeInfo.AllocatableResource()
+	requested := nodeInfo.RequestedResource()
+	var podRequest schedulernodeinfo.Resource
+	var podRequestInit schedulernodeinfo.Resource
+	switch resource {
```

Comment:
The three separate implementations seem duplicate. One way to unify them is to have a single function that computes the pod requested resources for a specific resource name:

```
calculateResourceRequest(pod *v1.Pod, resource v1.ResourceName) {
   var podRequest int64
   for i := range pod.Spec.Containers {
				container := &pod.Spec.Containers[i]
				value := priorityutil.GetNonzeroRequestForResource(resource, &container.Resources.Requests)
				podRequest += value
			}
			for i := range pod.Spec.InitContainers {
				container := &pod.Spec.InitContainers[i]
				value := priorityutil.GetNonzeroRequestForResource(resource, &container.Resources.Requests)
				if podRequest < value {
					podRequest = value
				}
			}
}
return podRequest
}
```

add GetNonzeroRequestForResource to util/non_zero.go that does something like this:

```
func GetNonzeroRequestForResource(resource v1.ResourceName, requests *v1.ResourceList) (int64, int64) {
  switch resource {
    case v1.ResourceCPU:
	if _, found := (*requests)[v1.ResourceCPU]; !found {
		return DefaultMilliCPURequest
	} else {
		return requests.Cpu().MilliValue()
	}
....
}
```

Reply:
This makes sense to me, also the function name could be GetLeastRequestForResource as it would return 0 for Scalar Resource or Ephemeral Storage. 👍